### ISSUE TITLE
can't just use auth_token to use metal things

### ISSUE THREAD

**Issue by cprivitere** [equinix/terraform-provider-equinix#122]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.

### Terraform Version

1.1.7

### Affected Resource(s)

* equinix_metal_*

### Terraform Configuration Files

```tf
terraform {
  required_providers {
    equinix = {
      source = "equinix/equinix"
      version = "1.6.0-alpha.2"
    }
  }
}

provider "equinix" {
  # Configuration options
  auth_token = var.auth_token
}

variable auth_token {
type = string
}
variable project_id {
type = string
}

resource "equinix_metal_device" "vmware" {
  hostname         = "vmware-a"
  plan             = "m3.large.x86"
  metro            = "da"
  operating_system = "vmware_esxi_7_0U2a"
  billing_cycle    = "hourly"
  project_id       = var.project_id
  tags             = ["vmware"]
}
```

### Expected Behavior

I should be able to provision equinix_metal resources.

### Actual Behavior

I get the following error:
```
❯ terraform plan
╷
│ Error: one of 'token' or pair 'client_id' - 'client_secret' must be set
│ 
│   with provider["registry.terraform.io/equinix/equinix"],
│   on main.tf line 10, in provider "equinix":
│   10: provider "equinix" {
│ 
```
It seems to require either token or client_id + client_secret, even though neither of those are used for equinix_metal provisioning.

### Steps to Reproduce

1. `terraform apply`

**Comment by displague** [comment]
The fix for this has not been fully released. It is available in v1.6.0-alpha.3

**Comment by btlogy** [comment]
I can confirm this also fix the need of having both METAL_AUTH_TOKEN and EQUINIX_API_TOKEN variables set in the environment (the 2nd being any non-empty string).

Thank you  @displague
